prod/prod.go

```diff
@@ -18,44 +18,108 @@ package prod
 
 import (
 	"context"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
 
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethersphere/swarm/chunk"
 	"github.com/ethersphere/swarm/pss"
 	"github.com/ethersphere/swarm/pss/trojan"
+	"github.com/ethersphere/swarm/storage/feed"
+	"github.com/ethersphere/swarm/storage/feed/lookup"
 )
 
-// RecoveryHook defines code to be executed upon trigger of failed to be retrieved chunks
+// RecoveryTopicText is the string used to construct the recovery topics
+const RecoveryTopicText = "RECOVERY"
+
+// RecoveryTopic is the topic used for repairing globally pinned chunks
+var RecoveryTopic = trojan.NewTopic("RECOVERY")
+
+// ErrPublisher is returned when the publisher string cannot be decoded into bytes
+var ErrPublisher = errors.New("failed to decode publisher")
+
+// ErrPubKey is returned when the publisher bytes cannot be decompressed as a public key
+var ErrPubKey = errors.New("failed to decompress public key")
+
+// ErrFeedLookup is returned when the recovery feed cannot be successefully looked up
+var ErrFeedLookup = errors.New("failed to look up recovery feed")
+
+// ErrFeedContent is returned when there is a failure to retrieve content from the recovery feed
+var ErrFeedContent = errors.New("failed to get content for recovery feed")
+
+// ErrTargets is returned when there is a failure to unmarshal the feed content as a trojan.Targets variable
+var ErrTargets = errors.New("failed to unmarshal targets in recovery feed content")
+
+// RecoveryHook defines code to be executed upon failing to retrieve pinned chunks
 type RecoveryHook func(ctx context.Context, chunkAddress chunk.Address) error
 
-// sender is the function call for sending trojan chunks
-type sender func(ctx context.Context, targets [][]byte, topic trojan.Topic, payload []byte) (*pss.Monitor, error)
+// sender is the function type for sending trojan chunks
+type sender func(ctx context.Context, targets trojan.Targets, topic trojan.Topic, payload []byte) (*pss.Monitor, error)
 
 // NewRecoveryHook returns a new RecoveryHook with the sender function defined
-func NewRecoveryHook(send sender) RecoveryHook {
+func NewRecoveryHook(send sender, handler feed.GenericHandler) RecoveryHook {
 	return func(ctx context.Context, chunkAddress chunk.Address) error {
-		targets, err := getPinners(chunkAddress)
+		targets, err := getPinners(ctx, handler)
```

Comment:
getPinners should be called when you get from manifest entry to context and here just extract targets from the  context  directly

Reply:
documented in #2181